### PR TITLE
Fix/newtoolchain opam2

### PR DESCRIPTION
**Pull request by jkrauska**
New toolchain build
Use opam.exports to decide what libs to install
Docker simplification
Remove gcr references (replace with docker hub)
Upgrade to ocamlformat 0.8 and reformat code.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -79,45 +78,31 @@ check-format:
 ########################################
 ## Containers and container management
 
+# customized local docker
+# fixme 'nanotest'
```
Comment (cmr) [inline Makefile]: this fixme was fixed right?

File: scripts/buildwarns.py [hunk 2]
```diff
@@ -29,32 +29,43 @@
...
+    except KeyError:
+        continue
 print '=' * 80
 print 'Warnings by file'
 for key, value in sorted(perfile.iteritems(), reverse=True, key=lambda (k,v): (v,k)):
-    print value, key
-    
+    try:
+        print value, key
+    except KeyError:
```
Comment (cmr) [inline scripts/buildwarns.py]: this and the next try/except don't do anything. print won't fail with a KeyError :)

File: dockerfiles/Dockerfile-toolchain [hunk 3]
```diff
@@ -28,37 +28,40 @@ RUN export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" && \
...
+#     num \
+#     ocamlformat \
+#     ocp-indent \
+#     orocksdb \
+#     ppx_deriving \
+#     ppx_deriving_yojson \
+#     rpc_parallel \
+#     utop \
+#     virtual_dom \
+#     yojson
```
Comment (bkase) [inline dockerfiles/Dockerfile-toolchain]: can we erase these instead of commenting them out

File: scripts/buildwarns.py (hunk 2 above)
Comment (jkrauska) [reply scripts/buildwarns.py]: ack, rm'd

File: src/app/lite/copy.ml [hunk 4]
```diff
@@ -24,28 +24,30 @@ let proof_tooltip_alt =
...
+      \             Namely, they need a merkle-path from the protocol state's \
+       ledger hash to their account."
   ; `New_line
   ; `Text
       "The succinct blockchain, protocol state, merkle-path, and account \
-       information are altogether just a few kilobytes,\n            \
-       so Coda can provide a full proof of the state of an account with just \
-       this tiny amount of data." ]
+       information are altogether just a few kilobytes,\n\
+      \            so Coda can provide a full proof of the state of an \
```
Comment (bkase) [inline src/app/lite/copy.ml]: can you remove these extra backslashes (I think the ocamlformat barfed here -- the website will look weird if these remain)

File: src/app/website/src/copy.ml [hunk 5]
```diff
@@ -31,9 +31,10 @@ let coda_slide_2 =
      its users, even at scale." ]
 
 let conclusion =
-  [ "Below is the live state of the Coda protocol testnet.\n            \
-     Your browser is fully verifying the state's integrity, and is essentially\n            \
-     acting as a full node."
+  [ "Below is the live state of the Coda protocol testnet.\n\
+    \            Your browser is fully verifying the state's integrity, and \
+     is essentially\n\
+    \            acting as a full node."
```
Comment (bkase) [inline src/app/website/src/copy.ml]: and here

File: src/app/website/src/main.ml [hunk 6]
```diff
@@ -287,11 +289,12 @@ let home () =
...
-                     the whole\n                \
-                     blockchain down to the size of a few tweets."
-                  ; "No one needs to store\n                \
-                     or download transaction history in order to verify the \
-                     blockchain." ])
+                     the whole\n\
+                    \                blockchain down to the size of a few \
+                     tweets."
+                  ; "No one needs to store\n\
+                    \                or download transaction history in order \
```
Comment (bkase) [inline src/app/website/src/main.ml]: and here

File: src/app/website/src/main.ml [hunk 7]
```diff
@@ -313,12 +316,12 @@ let home () =
...
-                     reliable software through the use of statically-typed \
-                     functional programming languages."
-                  ; "This is reflected in our OCaml codebase and style of\n                \
-                     structuring code around DSLs, as well as in the design \
-                     of languages we're developing for Coda." ])
+                  [ "A cornerstone of our approach is a focus on building\n\
+                    \                reliable software through the use of \
+                     statically-typed functional programming languages."
+                  ; "This is reflected in our OCaml codebase and style of\n\
+                    \                structuring code around DSLs, as well as \
```
Comment (bkase) [inline src/app/website/src/main.ml]: and here

File: src/app/website/src/main.ml [hunk 8]
```diff
@@ -338,11 +341,11 @@ let home () =
...
-                  [ "We implement state-of-the-art\n                \
-                     consensus protocols and have developed\n                \
-                     frameworks for describing distributed systems, enabling \
-                     us to quickly\n                \
-                     iterate." ])
+                  [ "We implement state-of-the-art\n\
+                    \                consensus protocols and have developed\n\
+                    \                frameworks for describing distributed \
+                     systems, enabling us to quickly\n\
+                    \                iterate." ])
```
Comment (bkase) [inline src/app/website/src/main.ml]: and here

**Pull request merged**